apps/prairielearn/src/ee/auth/lti13/lti13Auth.ts

```diff
@@ -86,18 +83,27 @@ router.post(
     // Name checking, not an error
     // LTI 1.3 spec defines sharing name as a MAY https://www.imsglobal.org/spec/lti/v1p3#users-and-roles
     // but discourages (MUST NOT) using other attributes for unique identifier
-    name = null;
+    let name: string | null = null;
     if (lti13_instance.name_attribute) {
       // Uses lodash.get to expand path representation in text to the object, like 'a[0].b.c'
       // Reasonable default is "name"
       // Points back to OIDC Standard Claims https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
       name = _get(lti13_claims, lti13_instance.name_attribute);
     }
 
+    let email: string | null = null;
+    if (lti13_instance.email_attribute) {
+      // Uses lodash.get to expand path representation in text to the object, like 'a[0].b.c'
+      // Reasonable default is "email"
+      // Points back to OIDC Standard Claims https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+      email = _get(lti13_claims, lti13_instance.email_attribute);
+    }
```

Comment:
@trombonekenny I patterned this off of the `name` attribute handling, but I'm wondering if we want to be stricter like we are with `uin` where we error out if the attribute isn't defined? What do you think?

Reply:
I think LTI 1.3 should be consistent with the other auth approaches in terms of what we consider optional and required. It would be weird if it got different attributes from different sources?

Reply:
I'm definitely in favor of requiring email from LTI 1.3. However, I do have a slight worry that some university might refuse to hand over emails for some crazy reason and we'll have to deal with that.

Reply:
I agree with @mwest1066. It's OK and reasonable to ask for email with LTI (and SAML and...) but we should be permissive in what schools store and send. Design-wise, if we really want emails we should likely ask/confirm the email the user wants to use than make assumptions (for example, people who don't check their school email).

Reply:
Could we assert that an email is present only if `lti13_instance.email_attribute` is defined? That way we don't require institutions to configure the attribute, but if it is set, we can enforce that we always get a value.

Or would y'all prefer to be super lenient here?

Reply:
I think your proposal is reasonable. We just want a way for institutions to opt out of needing to send emails. If they opt in then we can enforce it.